common/debugScreen.h

```diff
@@ -6,162 +6,152 @@
 #include <stdarg.h>
 #include <inttypes.h>
 
-#include <psp2/display.h>
-#include <psp2/kernel/sysmem.h>
-#include <psp2/kernel/threadmgr.h>
+typedef struct PsvDebugScreenFont {
+	unsigned char* glyphs, width, height, first, last, size_w, size_h;
+} PsvDebugScreenFont;
 
 #include <debugScreenFont.c>
 
 #define SCREEN_WIDTH    (960)
 #define SCREEN_HEIGHT   (544)
 #define SCREEN_FB_WIDTH (960)
-#define SCREEN_FB_SIZE  (2 * 1024 * 1024)
-#define SCREEN_FB_ALIGN (256 * 1024)
-#define SCREEN_GLYPH_W  (8)
-#define SCREEN_GLYPH_H  (8)
-#define SCREEN_TAB_SIZE (4)
-#define SCREEN_TAB_W    (SCREEN_GLYPH_W * SCREEN_TAB_SIZE)
+#define SCREEN_FB_SIZE  (2 * 1024 * 1024) //Must be 256KB aligned
+#ifndef SCREEN_TAB_SIZE /* this allow easy overriding */
+#define SCREEN_TAB_SIZE (8)
+#endif
+#define SCREEN_TAB_W    ((F.size_w) * SCREEN_TAB_SIZE)
+#define F psvDebugScreenFont
 
-#define COLOR_BLACK      0xFF000000
-#define COLOR_RED        0xFF0000FF
-#define COLOR_GREEN      0xFF00FF00
-#define COLOR_YELLOW     0xFF00FFFF
-#define COLOR_BLUE       0xFFFF0000
-#define COLOR_MAGENTA    0xFFFF00FF
-#define COLOR_CYAN       0xFFFFFF00
-#define COLOR_WHITE      0xFFFFFFFF
-#define COLOR_GREY       0xFF808080
```

Comment:
can you explain why remove these color macros?

Reply:
I removed the 2 non-standards `psvDebugSet{Bg/Fg}` function because `printf` can now set the fg/bg in a portable way.

Those COLOR macro were only needed by `psvDebugSet{Bg/Fg}` and so, are now useless.

If peoples still want to set they bg/fg using non-portable uint32_t color, they can use a wrappers like the one I made in the icmp sample. In that case, the macro definition belong to they code.

I used the debug_print code to check that all CSI sequence were correctly executed, but also to test if the font builder and the dynamic font swapping were correctly handled (hence the ASCII table dump).
Should rename this sample `debugScreen_demo` otherwise, what part should I strip ?

The primary goal for the http client was to demonstrate that BSD socket can be used instead of non-portable sceNet* fonction. So `bsd_socket` seems good :)

Reply:
btw, `COLOR_RED` is easier than `0xFF0000FF`. hm...

other things, original `debug_print` looks just print console messages, but this example is much more.
so I suggest making the new directory(yeah, like `bsd_socket` or another something) then copy these codesets.(I mean just make new sample not move. :) )
and IMO, rename to `debugscreen_demo` is also good.

Reply:
Done :)